packages/salesforcedx-vscode-core/src/orgBrowser/metadataOutlineProvider.ts

```diff
@@ -24,33 +37,144 @@ export class MetadataOutlineProvider
         const org = new BrowserNode(this.defaultOrg, NodeType.Org);
         return Promise.resolve([org]);
       } else {
-        return Promise.resolve([]);
+        const emptyDefault = new BrowserNode(
+          'No default org set',
+          NodeType.EmptyNode
+        );
+        return Promise.resolve([emptyDefault]);
+      }
+    } else if (element.type === NodeType.Org) {
+      const metadataTypes = await this.getTypes();
+      element.children = metadataTypes;
+      return Promise.resolve(element.children);
+    } else if (element.type === NodeType.MetadataType) {
+      const metadataCmps = await this.getComponents(element);
+      if (isNullOrUndefined(metadataCmps) || metadataCmps.length === 0) {
+        const emptyDefault = new BrowserNode(
+          'No components available',
+          NodeType.EmptyNode
+        );
+        element.children = [emptyDefault];
+      } else {
+        element.children = metadataCmps;
       }
-    } else {
       return Promise.resolve(element.children);
     }
+
+    return Promise.resolve([]);
   }
 
-  public async getDefaultUsernameOrAlias() {
+  public async getTypes(): Promise<BrowserNode[]> {
+    const username = this.defaultOrg!;
+    const typesFolder = await getTypesFolder(username);
+    const typesPath = path.join(typesFolder, 'metadataTypes.json');
+
+    let typesList: string[];
+    if (!fs.existsSync(typesPath)) {
+      try {
+        const result = await forceDescribeMetadata(typesFolder);
+        typesList = buildTypesList(result, undefined);
+      } catch (e) {
+        const errorNode = new BrowserNode(parseErrors(e), NodeType.EmptyNode);
+        return [errorNode];
+      }
+    } else {
+      typesList = buildTypesList(undefined, typesPath);
+    }
+    const nodeList = [];
+    for (const type of typesList) {
+      const typeNode = new BrowserNode(type, NodeType.MetadataType);
+      nodeList.push(typeNode);
+    }
+    return nodeList;
+  }
+
+  public async getComponents(
+    metadataType: BrowserNode
+  ): Promise<BrowserNode[]> {
+    const username = this.defaultOrg!;
+    const componentsPath = await getComponentsPath(
+      metadataType.label!,
+      username
+    );
+
+    let componentsList: string[];
+    if (!fs.existsSync(componentsPath)) {
+      const result = await forceListMetadata(
+        metadataType.label!,
+        username,
+        componentsPath
+      );
+      componentsList = buildComponentsList(
+        metadataType.label!,
+        result,
+        undefined
+      );
+    } else {
+      componentsList = buildComponentsList(
+        metadataType.label!,
+        undefined,
+        componentsPath
+      );
+    }
+    const nodeList = [];
+    for (const component of componentsList) {
+      const cmpNode = new BrowserNode(component, NodeType.MetadataCmp);
+      nodeList.push(cmpNode);
+    }
+    return nodeList;
+  }
+
+  public async getDefaultUsernameOrAlias(): Promise<boolean> {
```

Comment:
The name and responsibility of this method is a bit confusing. We use it to determine if the user has switched to a different org and also to set the username for the rest of the org browser code to leverage. Can we break this into different methods and name them based on what they are responsible for ?